course/viewbase.py

```diff
@@ -81,20 +85,23 @@ def get_resource_objects(self):
                 instance_def_language = instance_languages[0]
                 instance_languages = set(instance_languages)
 
+                languages = []
+                if self.user_course_data and self.user_course_data.language:
+                    languages.append(self.user_course_data.language)
+                if is_real_user and user.userprofile.language:
+                    languages.append(user.userprofile.language)
+
                 query_language = self.request.GET.get('hl')
                 if query_language:
                     if query_language[:2] in instance_languages:
                         language = query_language
+                        if languages:
+                            self.user_language = languages[0]
+                            if self.user_language[:2] != query_language[:2]:
+                                self.query_language = query_language
                     else:
                         raise TranslationNotFound
                 else:
-                    languages = []
-                    if self.user_course_data and self.user_course_data.language:
-                        languages.append(self.user_course_data.language)
-                    if is_real_user and user.userprofile.language:
-                        languages.append(user.userprofile.language)
-                    languages.append(get_language())
```

Comment:
This line is removed, so there is no cases when it would have meaning?

Reply:
Not sure, removing was probably accident